lib/base-compiler.js

```diff
@@ -172,12 +179,27 @@ Compile.prototype.compile = function (source, options, filters) {
                         result.asm = "<Compilation failed>";
                         return result;
                     }
+                    result.hasOptOutput = false;
+                    if(self.couldHaveOptOutput(options, self.compiler.version)) {
+                        const optPath = path.join(dirPath, outputFilebase + ".opt.yaml");
```

Comment:
i thought i had removed all of these, noticed that there were none of the other more modern JS features. Although const is pretty safe.

It might be good to specify the current LTS as a minimum version?